extensions/amp-experiment/1.0/mutation-parser.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {isObject} from '../../../src/types';
+import {userAssert} from '../../../src/log';
+
+/**
+ * Function to find all selectors of the mutation
+ * and return a function to apply the identified
+ * MutationRecord
+ * @param {!JsonObject} mutation
+ * @param {!Document} document
+ * @return {!Function}
+ */
+export function parseMutation(mutation, document) {
+
+  const mutationRecord = assertMutationRecord_(mutation);
```

Comment:
I'm fine with us having different function at this time given we have a tight deadline. 

But once we have time, what do you think of creating different mutationRecord class for each. Similar to the AMP Analytics `EventTracker`. I think that would help us to keep code cleaner depending on how much logic each type of mutation share.

Reply:
Ahh so similar to [my benchmarking implementation](https://github.com/torch2424/amp-benchmark/blob/master/benchmark/mutationRecords.js)? Yes I can go ahead and do that! After AMP Conf. Want me to just go ahead and make a `amp-experiment: Post AMP Conf Tasks` issue? 😄 

That being said, the mutations do share a bit of logic, but with the current implementation, I don't see there being any code copy-paste if that makes sense?

If anything we can discuss offline, but I am okay going with either way 👍

Reply:
> Ahh so similar to my benchmarking implementation? Yes I can go ahead and do that! After AMP Conf. Want me to just go ahead and make a amp-experiment: Post AMP Conf Tasks issue? 😄

Yes please, a task tracker will be super helpful. And right similar to your benchmarking implementation, except that I am thinking about using abstract base class. 
Once we have the real mutation logic, we will have more context on which is the better approach : )

Reply:
Set this up at https://github.com/ampproject/amphtml/issues/21705 😄